### PR TITLE
OCLOMRS-799:Implements retrieving, creating and searching orgs

### PR DESCRIPTION
**Pull request by hadijahkyampeire**
# JIRA TICKET NAME:
[As an organization admin, I want to list organizations so that I can create content under those organizations
](https://issues.openmrs.org/browse/OCLOMRS-799)

# Summary:
## Subticket: https://issues.openmrs.org/browse/OCLOMRS-803
- Viewing user organisations.
- Creating an organisation.
- Searching through organisations.
- And some styling of the selected/active link of the `NavDrawer`

### CODE REVIEW

**Comment by dkayiwa** [comment]
Can you include the ticket id in the commit message as recommended here? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

File: src/apps/organisations/components/OrganisationForm.tsx
Comment (ibacher) [inline src/apps/organisations/components/OrganisationForm.tsx:33]:
```suggestion
  website: Yup.string().url().notRequired(),
```

File: src/apps/organisations/components/OrganisationForm.tsx
Comment (ibacher) [inline src/apps/organisations/components/OrganisationForm.tsx:153]:
```suggestion
                <MenuItem value="View">View</MenuItem>
                <MenuItem value="Edit">Edit</MenuItem>
                <MenuItem value="None">None</MenuItem>
```

File: src/apps/organisations/components/OrganisationForm.tsx
Comment (ibacher) [inline src/apps/organisations/components/OrganisationForm.tsx:45]:
```suggestion
  public_access: "View",
```

File: src/apps/organisations/components/OrganisationForm.tsx
Comment (ibacher) [inline src/apps/organisations/components/OrganisationForm.tsx:24]:
```suggestion
}

const ExtrasSchema = Yup.object().shape<Extras>({
```

Comment (ibacher) [inline src/apps/organisations/components/OrganisationForm.tsx:24]: I think we can dispense with having a whole type and schema defined for `extras` when we're really just treating it as a pass-through string. I actually would be inclined to keep this out of the UI altogether until we have a definitive reason for reading or writing this metadata. The `ExtrasSchema` we use in the `concept` app is actually used to store a range of metadata important to OpenMRS, but which doesn't fit into the default OCL API (mostly because it only applies to numerical concepts). Here, even if OCL supports additional metadata, I don't think we actually use it for anything (and hand-editing a JSON document in a field is not a great idea).

File: src/apps/organisations/components/OrganisationForm.tsx
Comment (ibacher) [inline src/apps/organisations/components/OrganisationForm.tsx:172]:
```suggestion
              <Link to={`/user/orgs/`}>
```

File: src/apps/organisations/components/ViewOrganisations.tsx
Comment (ibacher) [inline src/apps/organisations/components/ViewOrganisations.tsx:15]: While it seems to work that typing, e.g., "CIEL" into the search box changes the URL to `http://localhost:8080/user/organisations/?q=CIEL`, if I browse to this URL directly (after logging in) the search field is not filled out.

In other:

1. Login to OCL for OpenMRS
2. Navigate to `http://localhost:8080/user/organisations/?q=CIEL`
3. Note that the search box is blank even though the search *results* are actually filtered

I'm not sure exactly what the cause is here, but we need to take a look into that.

File: src/apps/organisations/components/index.ts
Comment (ibacher) [inline src/apps/organisations/components/index.ts:2]:
```suggestion
export { default as OrganisationForm } from "./OrganisationForm";

```

File: src/apps/organisations/components/organisationCard.tsx
Comment (ibacher) [inline src/apps/organisations/components/organisationCard.tsx:1]: nit: This file should be called `OrgCard.tsx` (or, at least `OrganisationCard.tsx`).

File: src/apps/organisations/components/organisationCard.tsx
Comment (ibacher) [inline src/apps/organisations/components/organisationCard.tsx:44]: Personally, I don't think there's a great deal of value to adding this to the card. It would, however, be useful to display the `id` of the organisation.

File: src/apps/organisations/pages/CreateOrganisationPage.tsx
Comment (ibacher) [inline src/apps/organisations/pages/CreateOrganisationPage.tsx:65]:
```suggestion
)(CreateOrganisationPage);

```

**Review by ibacher — CHANGES_REQUESTED** [review]
@hadijahkyampeire Thank you for your great work on this! I've added a number of comments mostly for minor things. There is one major thing I'd like to bring up, which is that I think it's easier if we shortedn the word `organisation` (when it's not displayed in the UI`) to `org` or `orgs` as appropriate, e.g. `OrgCard.tsx`, `ViewOrgs.tsx`, etc.

File: src/apps/organisations/components/ViewOrganisations.tsx
Comment (hadijahkyampeire) [reply src/apps/organisations/components/ViewOrganisations.tsx:15]: Yah, I saw the issue of the search field but I think to fix it I will have to do some changes to the existing Search Container and it is used in all search places, so that might be some work but I will see it I can fix it now or do it as a follow up PR.

File: src/apps/organisations/components/organisationCard.tsx
Comment (hadijahkyampeire) [reply src/apps/organisations/components/organisationCard.tsx:1]: Oh yah, I have been used to camelCase but I will fix that.

File: src/apps/organisations/components/OrganisationForm.tsx
Comment (hadijahkyampeire) [reply src/apps/organisations/components/OrganisationForm.tsx:24]: Sounds good

**Comment by hadijahkyampeire** [comment]
@dkayiwa please review this PR and we see if we can merge it because other tickets are being blocked by it.

**Comment by dkayiwa** [comment]
@hadijahkyampeire I have just given you write access to this repository such that you can merge pull requests like these, which  block others. In case of any review comments, after the pull request is merged, you can just create a new one.

**Comment by hadijahkyampeire** [comment]
Thanks @dkayiwa

**Pull request merged**
